src/uu/printf/src/printf.rs

```diff
@@ -46,6 +46,13 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
         };
     }
 
+    // See #5815 - We don't need to iter on args if no format string seen
+    let format_seen =
+        parse_spec_and_escape(format_string.as_ref()).any(|r| matches!(r, Ok(FormatItem::Spec(_))));
```

Comment:
Maybe we can move this check to the for loop above? Then we don't have to parse the format string twice. I'm thinking this just above the match:
```rust
if let Ok(FormatItem::Spec(_)) = item {
    format_seen = true;
}
```

Other small remark: I don't really like referring to (closed) issues from the code. Ideally, the reasoning should be clear without opening up GitHub. It'd be great if you could add a bit more context here to the comment. Something along the lines of "If there are not format specs in the format string, then we would not consume any arguments in the iteration below and we would get stuck in an infinite loop. Therefore we exit early."

Reply:
fixed. Thanks